packages/synchronize/lib/src/object.dart

```diff
@@ -0,0 +1,22 @@
+part of '../synchronize.dart';
+
+/// Wraps the actual data contained on each side.
+class SyncObject<T> {
```

Comment:
I always think that classes with only two members can also be represented as records.
```dart
typedef SyncObject<T> = (String, T);
```

What do you think? I'm not sure whether this makes sense here

Reply:
I think it makes a lot of sense. This type is only used to hold some data and has no logic so a class is not necessary.

Reply:
For SyncAction it's better to have it as the sealed class for exhaustive matching